README.rst

```diff
@@ -0,0 +1,210 @@
+====================
+CMake Common Scripts
+====================
+This contains a set useful cmake macros
+
+.. contents:: Table of Contents
+   :depth: 3
+
+
+Available Macros
+================
+
+Clang Tidy
+----------
+
+This CMake macro will add clang tidy to a provided target.
+
+Macro:
+
+.. code-block:: cmake
+
+   target_clang_tidy(${PACKAGE_NAME} ${ARGN})
+
+This configures clang-tidy with default arguments to only produce warnings.
+
+.. code-block:: cmake
+
+   target_clang_tidy(${PACKAGE_NAME} ${DEFAULT_CLANG_TIDY_WARNING_ARGS})
+
+This configures clang-tidy with default arguments to only produce errors.
+
+.. code-block:: cmake
+
+   target_clang_tidy(${PACKAGE_NAME} ${DEFAULT_CLANG_TIDY_ERROR_ARGS})
+
+Include What You Use (IWYU)
+---------------------------
+This CMake macro will add IWYU to a given target
+
+.. code-block:: cmake
+
+   target_include_what_you_use(${PACKAGE_NAME} ${ARGN})
+
+This CMake macro will add IWYU to a given target with default arguments.
+
+.. code-block:: cmake
+
+   target_include_what_you_use(${PACKAGE_NAME} ${DEFAULT_IWYU_ARGS})
+
+
+This CMake macro will add IWYU to all targets
+
+.. code-block:: cmake
+
+   include_what_you_use(${ARGN})
+
+
+This CMake macro will add IWYU to all targets with default arguments.
+
+.. code-block:: cmake
+
+   include_what_you_use(${DEFAULT_IWYU_ARGS})
+
+
+CppCheck
+--------
+
+This CMake macro will add CppCheck to a given target
+
+.. code-block:: cmake
+
+   target_cppcheck(${PACKAGE_NAME} ${ARGN})
+
+
+This CMake macro will add CppCheck to a given target with default arguments.
+
+.. code-block:: cmake
+
+   target_cppcheck(${PACKAGE_NAME} ${DEFAULT_CPPCHECK_ARGS})
+
+
+This CMake macro will add CppCheck to all targets
+
+.. code-block:: cmake
+
+   cppcheck(${ARGN})
+
+
+This CMake macro will add CppCheck to all targets with default arguments.
+
+.. code-block:: cmake
+
+   cppcheck(${DEFAULT_CPPCHECK_ARGS})
+
+
+Configure (Pure CMake Package)
+------------------------------
+This CMake macro simplifies the CMake package configure and install by performing multiple operations
+
+* It installs the provided targets
+* It exports the provided targets under the provided namespace
+* It installs the package.xml file
```

Comment:
Perhaps we should add installing (and generating?) colcon hooks?

Reply:
We can open that as an issue if you want to do that later

Reply:
I would not do this by default actually. From the discussion on the `tesseract` PR, I got the impression it should only be needed if ROS 2 packages want/need to be able to locate artefacts in this package at runtime.

CMake macros/functions are by definition used at build time.

Reply:
I asked and the colcon.pkg is only required at build time and does not need to be installed, though the hooks do need to be installed. They could be added here along side the package.

>

I would not do this by default actually. From the discussion on the tesseract PR, I got the impression it should only be needed if ROS 2 packages want/need to be able to locate artefacts in this package at runtime.

CMake macros/functions are by definition used at build time.


Are you recommending not to install the hooks and package.xml by default?